### PR TITLE
Change how character devices are registered

### PR DESCRIPTION
**Pull request by veidongray**
I changed the way I register character devices and used simpler functions to reduce the amount of code.

### CODE REVIEW

File: fibdrv.c [hunk 1]
```diff
@@ -20,9 +20,10 @@ MODULE_VERSION("0.1");
 #define MAX_LENGTH 92
 
 static dev_t fib_dev = 0;
-static struct cdev *fib_cdev;
 static struct class *fib_class;
 static DEFINE_MUTEX(fib_mutex);
+static int major = 0;
```
Comment (jserv) [inline fibdrv.c]: Combine the declaration of major and minor in a single line. i.e.,
```c
static int major = 0, minor = 0;
```

File: fibdrv.c [hunk 2]
```diff
@@ -107,34 +108,18 @@ const struct file_operations fib_fops = {
...
-    fib_cdev = cdev_alloc();
-    if (fib_cdev == NULL) {
-        printk(KERN_ALERT "Failed to alloc cdev");
-        rc = -1;
-        goto failed_cdev;
-    }
-    fib_cdev->ops = &fib_fops;
-    rc = cdev_add(fib_cdev, fib_dev, 1);
-
+    major = register_chrdev(major, DEV_FIBONACCI_NAME, &fib_fops);
```
Comment (jserv) [inline fibdrv.c]: Rewrite as following:
```c
rc = major = register_chrdev(major, DEV_FIBONACCI_NAME, &fib_fops);
```

File: fibdrv.c (hunk 1 above)
Comment (veidongray) [reply fibdrv.c]: Ok, you're right, thanks for the correction.

File: fibdrv.c (hunk 2 above)
Comment (veidongray) [reply fibdrv.c]: I'll rewrite this, thank you.

**Comment by jserv** [comment]
Thank @YIVIES for contributing!

**Pull request merged**